src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -83,29 +93,38 @@ object GroupReadsByUmi {
         tmp
       }
       else {
-        val lib       = library(rec)
-        val chrom     = rec.refIndex
-        val mateChrom = rec.mateRefIndex
-        val recNeg    = rec.negativeStrand
-        val recPos    = if (recNeg) rec.unclippedEnd else rec.unclippedStart
-
-        val (mateNeg, matePos) = if (!rec.paired) (false, Int.MaxValue) else {
-          val neg = rec.mateNegativeStrand
-          val pos = if (neg) SAMUtils.getMateUnclippedEnd(rec.asSam) else SAMUtils.getMateUnclippedStart(rec.asSam)
-          (neg, pos)
+        val lib = library(rec)
+
+        val (ref1, start1, strand1) = if (rec.unmapped) (UnknownRef, UnknownPos, UnknownStrand) else {
+          val start = if (rec.positiveStrand) rec.unclippedStart else rec.unclippedEnd
+          (rec.refIndex, start, strandToByte(rec.positiveStrand))
+        }
+
+        val (ref2, start2, strand2) = if (rec.unpaired || rec.mateUnmapped) (UnknownRef, UnknownPos, UnknownStrand) else {
+          val start = if (rec.matePositiveStrand) SAMUtils.getMateUnclippedStart(rec.asSam) else SAMUtils.getMateUnclippedEnd(rec.asSam)
```

Comment:
Worth it to add helpers on `SamRecord`? In parity with:

https://github.com/fulcrumgenomics/fgbio/blob/7a91c97350b308c35d054d5080a42482d0eff12d/src/main/scala/com/fulcrumgenomics/bam/api/SamRecord.scala#L151-L152

Reply:
Huh, turns out there are already `mateUnclippedStart` and `mateUnclippedEnd` functions there!